### PR TITLE
Renamed full diversity rx json configs to be more descriptive

### PR DESCRIPTION
**Pull request by David-OConnor**
Addresses https://github.com/ExpressLRS/ExpressLRS/discussions/1742

Bottom line: The json files for Rx targets don't include comments; it appears the file names are the delineators. Clarifies that these 2 targets are full diversity, and included PA.

### CODE REVIEW

**Review by pkendall64 — COMMENTED** [review]
We should probably expand the `product_name` as well so when it's displayed it's obvious as well.

**Comment by David-OConnor** [comment]
Done. Also modified names of all ESP32 rx targets to be inline with other tgts.

**Comment by David-OConnor** [comment]
Pluses: That's the convention used for many other generic targets. For example, the `"rx_2400"` one directly above those changed in the PR (But not all). In hindsight, we can standardize this in this PR. Would y'all prefer plusses, or not?

"Diversity" and "PA" ordering: Same thing; it's `PA Diversity` in layout files, and `Diversity PA` in `product name`s

**Comment by pkendall64** [comment]
Personally I prefer `[[Full ]Diversity] PA]` and it's a no for pluses from me too.

**Comment by David-OConnor** [comment]
Done; standardized `layout_file` and `product_name` re `PA`/`Diversity` order, and `+`.

File: src/hardware/targets.json [hunk 1]
```diff
@@ -224,15 +224,15 @@
                 "upload_methods": ["uart", "wifi", "betaflight"]
             },
             "diversity": {
-                "product_name": "Generic ESP32 2.4Ghz Diversity RX",
+                "product_name": "Generic ESP32 Full Diversity PA 2.4Ghz RX",
```
Comment (JyeSmith) [inline src/hardware/targets.json:227]: Minor, but we've been using "True Diversity" to differentiate these receivers from Antenna diversity.

What do we think about changing the 4 instances below from Full Diversity to True Diversity?

Comment (CapnBry) [reply src/hardware/targets.json:227]: _100_

**Comment by David-OConnor** [comment]
Done

**Review by pkendall64 — APPROVED** [review]

**Review by JyeSmith — APPROVED** [review]

**Comment by CapnBry** [comment]
This PR broke the targets using `board = generic.rx_2400.diversity` as there now are two things named that. :(

**Comment by pkendall64** [comment]
> This PR broke the targets using `board = generic.rx_2400.diversity` as there now are two things named that. :(

Added a fix for that in the fix for the happymodel dual PR #1872

**Pull request merged**